src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/src/Client/ElectricityMapsClient.cs

```diff
@@ -103,6 +94,72 @@ public async Task<ForecastedCarbonIntensityData> GetForecastedCarbonIntensityAsy
         return await GetCurrentForecastAsync(parameters);
     }
 
+    public async Task<PastRange> GetPastRangeDataAsync(string latitude, string longitude, DateTimeOffset startTime, DateTimeOffset endTime)
+    {
+        _log.LogDebug("Requesting carbon intensity using latitude {latitude} longitude {longitude}",
+           latitude, longitude);
+
+        var parameters = new Dictionary<string, string>()
+        {
+            { QueryStrings.Latitude, latitude },
+            { QueryStrings.Longitude, longitude },
+            { QueryStrings.StartTime, DateTimeToString(startTime) },
+            { QueryStrings.EndTime, DateTimeToString(endTime) },
+        };
+
+        return await GetPastRangeDataAsync(parameters);
+    }
+
+    public async Task<PastRange> GetPastRangeDataAsync(string zone, DateTimeOffset startTime, DateTimeOffset endTime)
+    {
+        _log.LogDebug("Requesting carbon intensity using zone {zone}",
+           zone);
+
+        var parameters = new Dictionary<string, string>()
+        {
+            { QueryStrings.ZoneName, zone },
+            { QueryStrings.StartTime, DateTimeToString(startTime) },
+            { QueryStrings.EndTime, DateTimeToString(endTime) },
+        };
+
+        return await GetPastRangeDataAsync(parameters);
+    }
+    
+    private static string DateTimeToString(DateTimeOffset dateTime)
+    {
+        return dateTime.ToUniversalTime().ToString("O", CultureInfo.InvariantCulture);
+    }
+
+    // Internal call to check for allowed zones and then make GET request to History endpoint
+    private async Task<HistoryCarbonIntensityData> GetHistoryCarbonIntensityDataAsync(Dictionary<string, string> parameters)
+    {
+        await CheckZonesAllowedForPath(Paths.History, parameters);
+        AddOptionalParameters(parameters);
+
+        using (var result = await this.MakeRequestGetStreamAsync(Paths.History, parameters))
+        {
+            return await JsonSerializer.DeserializeAsync<HistoryCarbonIntensityData>(result, _options) ?? throw new ElectricityMapsClientException($"Error getting history carbon intensity data");
+        }
+    }
+    private void AddOptionalParameters(Dictionary<string, string> parameters)
+    {
+        if (_configuration.EmissionFactorType != null)
+            parameters.Add(QueryStrings.EmissionFactorType, _configuration.EmissionFactorType);
+
+        if (_configuration.DisableEstimations != null)
+            parameters.Add(QueryStrings.DisableEstimations, _configuration!.DisableEstimations.ToString()!.ToLowerInvariant());
+    }
+    
+    private async Task<PastRange> GetPastRangeDataAsync(Dictionary<string, string> parameters)
+    {
+        await CheckZonesAllowedForPath(Paths.PastRange, parameters);
+        AddOptionalParameters(parameters);
+        using (var result = await this.MakeRequestGetStreamAsync(Paths.PastRange, parameters))
```

Comment:
keyword using can be simplified with one liner, instead of using brackets to keep the scope. See `GetCurrentForecastAsync` method